src/wrapper/jnienv.rs

```diff
@@ -100,17 +100,34 @@ impl<'a> JNIEnv<'a> {
         Ok(jni_unchecked!(self.internal, GetVersion).into())
     }
 
-    /// Define a new java class. See the JNI docs for more details - I've never
-    /// had occasion to use this and haven't researched it fully.
+    /// Load a class from a buffer of raw class data. The name of the class must match the name
+    /// encoded within the class file data.
     pub fn define_class<S>(&self, name: S, loader: JObject<'a>, buf: &[u8]) -> Result<JClass<'a>>
     where
         S: Into<JNIString>,
     {
         let name = name.into();
+        self.define_class_impl(name.as_ptr(), loader, buf)
+    }
+
+    /// Load a class from a buffer of raw class data.
```

Comment:
Shall we add that the VM _infers_ the class name from the buffer, as the issue suggests? As "unnamed class" might sound confusing.

Reply:
I have updated the documentation, but I agree with you that `define_unnamed_class` isn't the best name for the method. I suppose it would be better to have `define_class` and `define_named_class` functions, but it would be a breaking change.